elementary/monitor/dbt_project/macros/alerts/get_model_alerts.sql

```diff
@@ -0,0 +1,127 @@
+{% macro get_pending_model_alerts(days_back) %}
+    -- depends_on: {{ ref('alerts_models') }}
+    {% set elementary_database, elementary_schema = elementary.target_database(), target.schema %}
+    {% set snapshots_relation = adapter.get_relation(elementary_database, elementary_schema, 'dbt_snapshots') %}
+
+
+    {% set select_pending_alerts_query %}
+        with alerts_in_time_limit as (
+            select *
+            from {{ ref('alerts_models') }}
+            where {{ elementary.cast_as_timestamp('detected_at') }} >= {{ get_alerts_time_limit(days_back) }}
+        ),
+
+        models as (
+            select * from {{ ref('elementary', 'dbt_models') }}
+        ),
+
+        {% if snapshots_relation %}
+            snapshots as (
+                select * from {{ snapshots_relation }}
+            ),
+        {% endif %}
+
+        artifacts_meta as (
+            select unique_id, meta from models
+            {% if snapshots_relation %}
+                union all
+                select unique_id, meta from {{ snapshots_relation }}
+            {% endif %}
+        )
+        select
+            alerts_in_time_limit.alert_id,
+            alerts_in_time_limit.unique_id,
+            alerts_in_time_limit.detected_at,
+            alerts_in_time_limit.database_name,
+            alerts_in_time_limit.materialization,
+            alerts_in_time_limit.path,
+            alerts_in_time_limit.original_path,
+            alerts_in_time_limit.schema_name,
+            alerts_in_time_limit.message,
+            alerts_in_time_limit.owners,
+            alerts_in_time_limit.tags,
+            alerts_in_time_limit.alias,
+            alerts_in_time_limit.status,
+            alerts_in_time_limit.full_refresh,
+            {# backwards compatibility #}
+            case
+                when alerts_in_time_limit.suppression_status is NULL and alerts_in_time_limit.alert_sent = TRUE then 'sent'
+                when alerts_in_time_limit.suppression_status is NULL and alerts_in_time_limit.alert_sent = FALSE then 'pending'
+                else suppression_status
+            end as suppression_status,
+            case 
+                when alerts_in_time_limit.sent_at is NULL then '1970-01-01 00:00:00'
+                else alerts_in_time_limit.sent_at
+            end as sent_at,
```

Comment:
Why do we need to keep track of when was the alert sent?

Reply:
The suppression is calculated by the time the alert sent, and not the time the alert detected at (product definition which IMO is right) 🙂